File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}
        for i in range(0, len(nums)):
            if nums[i] in mp:
                return [mp[nums[i]], i]
            mp[target - nums[i]] = i

        return []

File: Data Structures & Algorithms/subsets/submission-0.java
class Solution {
    public void createSubset(int index, int[] nums, List<Integer> stack, List<List<Integer>> sol){
        if(index == nums.length){
            sol.add(new ArrayList<>(stack));
        } else if (index < nums.length){
            stack.add(nums[index]);
            createSubset(index + 1, nums, stack, sol);
            stack.remove(stack.size() - 1);
            createSubset(index + 1, nums, stack, sol);
        }
    }

    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> sol = new ArrayList<>();
        List<Integer> stack = new ArrayList<>();

        createSubset(0, nums, stack, sol);
        return sol;
    }
}
